addons/account/static/src/js/section_and_note.js

```diff
@@ -0,0 +1,86 @@
+/*
+    The goal of this file is to contain JS hacks related to allowing
+    section and note on sale order and invoice.
+*/
+odoo.define('account.list_renderer_section_and_note', function (require) {
+
+"use strict";
+var ListRenderer = require('web.ListRenderer');
+var fieldRegistry = require('web.field_registry');
+var FieldOne2Many = require('web.relational_fields').FieldOne2Many;
+
+var SectionAndNoteListRenderer = ListRenderer.extend({
+    // We don't want to select the cell for note and section, we only
+    // want to edit them with the dialog because they have additional fields.
+    _onCellClick: function (ev) {
+        var currentLine = $(ev.currentTarget.parentNode);
+
+        var isNoteOrSection = currentLine && (
+            currentLine.hasClass('is-line_note')
+            || currentLine.hasClass('is-line_section')
+        );
+
+        if (isNoteOrSection) {
+            return;
+        }
+
+        this._super.apply(this, arguments);
+    },
+    // We only want dialog edit for note and section.
+    _onRowClicked: function (ev) {
+        var currentLine = $(ev.currentTarget);
+
+        var isNoteOrSection = currentLine && (
+            currentLine.hasClass('is-line_note')
+            || currentLine.hasClass('is-line_section')
+        );
+
+        var args = Array.prototype.slice.call(arguments);
+        args.push(isNoteOrSection);
+        this._super.apply(this, args);
+    },
+    // For section or note the "name" column is the only one that should
+    // be displayed, between the handle and the delete button.
+    _renderBodyCell: function (record, node, index, options) {
+        var $cell = this._super.apply(this, arguments);
+
+
+        var isSection = record.data['line_type'] === 'line_section';
+        var isNote = record.data['line_type'] === 'line_note';
+
+        if (isSection || isNote) {
+            if ($cell.hasClass('o_handle_cell')) {
+                return $cell;
+            } else if ($cell.hasClass('product_name')) {
+                var nbrColumns = this._getNumberOfCols();
+                nbrColumns = this.addTrashIcon ? nbrColumns - 2 : nbrColumns;
+                $cell.attr('colspan', nbrColumns);
+            } else {
+                return $cell.addClass('o_hidden');
+            }
+        }
+
+        return $cell;
+    },
+    // We add the is-{type} class to allow custom behaviour both in JS and CSS.
+    _renderRow: function (record, index) {
+        var $row = this._super.apply(this, arguments);
+
+        $row.addClass('is-' + record.data['line_type']);
```

Comment:
type out the class manually for easier grep?

what name should it be? add _o ? everything underscore?